src/Microsoft.OData.Client/BulkUpdateSaveResult.cs

```diff
@@ -184,13 +279,23 @@ private ODataRequestMessageWrapper GenerateBulkUpdateRequest()
             using (ODataMessageWriter messageWriter = Serializer.CreateDeltaMessageWriter(bulkUpdateRequestMessage, this.RequestInfo, isParameterPayload: false))
             {
                 this.writer = ODataWriterWrapper.CreateForDeltaFeed(messageWriter, this.bulkUpdateGraph.EntitySetName, this.RequestInfo.Configurations.RequestPipeline, bulkUpdateRequestMessage, this.RequestInfo);
-                this.SerializerInstance.WriteDeltaResourceSet(this.bulkUpdateGraph.TopLevelDescriptors, this.bulkUpdateGraph, writer);
+                this.SerializerInstance.WriteDeltaResourceSet(this.bulkUpdateGraph.TopLevelDescriptors, this.bulkUpdateGraph, this.linkDescriptors, writer);

```

Comment:
If we're passing `bulkUpdateGraph`, why do we need to pass `bulkUpdateGraph.TopLevelDescriptors`? Why can't we access the `TopLevelDescriptors` property from `WriteDeltaResourceSet` method?

nit: If we need to pass `bulkUpdateGraph` and top-level descriptors separately, might it be cleaner to refactor `WriteDeltaResourceSet` method to accept `BulkUpdateGraph, IReadOnlyList<Descriptor>, Dictionary<Descriptor, HashSet<LinkDescriptor>>, ODataWriterWrapper)` in that order, i.e., descriptors collections adjacent to each other.
`WriteDeltaEntry` and `WriteDeltaNestedResourceInfo` would also look cleaner if the parameters were ordered `BulkUpdateGraph, EntityDescriptor, Dictionary<Descriptor, HashSet<LinkDescriptor>>, ODataWriterWrapper` in that order. It's not a deal breaker though. It just looks cleaner - at least to the OCD me!

Reply:
The reason we pass `bulkUpdateGraph.TopLevelDescriptors` is because the `WriteDeltaResourceSet` is called recursively. This means that when we are writing the nested resources for example we need to pass in the list of nested objects that are being written instead of passing the bulkupdategraph and starting to traverse it to figure out what needs to be written.

Reply:
Have taken  this order: `IReadOnlyList<Descriptor>, Dictionary<Descriptor, HashSet<LinkDescriptor>>, BulkUpdateGraph, ODataWriterWrapper)`